zerver/lib/events.py

```diff
@@ -661,6 +661,10 @@ def name(sub: Dict[str, Any]) -> str:
                 if (sub['stream_id'] == stream_id and
                         user_id in sub['subscribers']):
                     sub['subscribers'].remove(user_id)
+            for sub in state['unsubscribed']:
+                if (sub['stream_id'] == stream_id and
+                        user_id in sub['subscribers']):
+                    sub['subscribers'].remove(user_id)
```

Comment:
It looks like the `peer_add` code path also ignores `unsubscribed`?  

In any case, givne that the code is the same in these inner loops, can we just do:

`for sub in state['subscribed'] + state['unsubscribed'] + state['never_subscribed']`?